sparkle/platform/output.py

```diff
@@ -0,0 +1,287 @@
+        self.training, self.best_config =\
+            self.get_validation_data(self.instance_set_train)
+
+        # Retrieve test validation results if they exist
+        if self.instance_set_test is not None:
+            self.test, _ = self.get_validation_data(self.instance_set_test)
+
+    def parse_string_to_dict(self: ConfigurationOutput, input_string: str) -> dict:
+        """Convert a configuration string to a dictionary."""
+        result_dict = {}
+        input_string = input_string.replace("'", "")
+        input_string = input_string.replace(",", "")
+        input_string = input_string.replace(":", "")
+        input_string = input_string.replace("{", "")
+        input_string = input_string.replace("}", "")
+        tokens = input_string.split()
+
+        for i in range(0, len(tokens), 2):
+            key = tokens[i]
+            value = tokens[i + 1]
+            result_dict[key] = value
+
+        return result_dict
+
+    def get_validation_data(self: ConfigurationOutput, instance_set: InstanceSet) -> \
+            tuple[ConfigurationResults, str]:
+        """Returns best config and ConfigurationResults for instance set."""
+        objective = self.configurator.scenario.sparkle_objective
+
+        # Retrieve found configuration
+        _, best_config = self.configurator.get_optimal_configuration(
+            self.solver, instance_set, objective.PerformanceMeasure)
+
+        # Retrieve validation results
+        validator = Validator(self.directory)
+        val_default = validator.get_validation_results(
+            self.solver, instance_set,
+            source_dir=self.directory, subdir="validation")
+        val_conf = validator.get_validation_results(
+            self.solver, instance_set, config=best_config,
+            source_dir=self.directory, subdir="validation")
+
+        results = []
+        # Form: 0: solver, 1: config, 2: set, 3: instance, 4: status,
+        # 5: quality, 6: runtime
+        for res in val_default:
+            # TODO: status to enum
+            results.append([res[3], res[4], res[5], res[6]])
+        configuration = self.parse_string_to_dict(val_default[0][1])
+        results_default = ValidationResults(self.solver, configuration,
+                                            instance_set, results)
+
+        results = []
+        # Form: 0: solver, 1: config, 2: set, 3: instance, 4: status,
+        # 5: quality, 6: runtime
+        for res in val_conf:
+            # TODO: status to enum
+            results.append([res[3], res[4], res[5], res[6]])
+        configuration = self.parse_string_to_dict(val_conf[0][1])
+        results_conf = ValidationResults(self.solver, configuration,
+                                         instance_set, results)
+
+        cutoff_time = self.configurator.scenario.cutoff_time
+        penalty = penalty_multiplier * \
+            self.configurator.scenario.cutoff_time
+        perf_par_conf = sgrfch.get_par_performance(val_default,
```

Comment:
preferably don't use methods from the generate report methods, as this should drive the report generation afterwards (And the code in those files is quite shit)